Example/Tests/UserTests.swift

```diff
@@ -177,43 +176,86 @@ class UserTests: QuickSpec {
-                _ = user.didSetTokens.register { newTokens = $0.new }
+                try? user.set(accessToken: "hehe", refreshToken: "hehe", idToken: "hehe", userID: "hehe")
+                try? user.set(accessToken: "hehe again")
+                expect(user.tokens).toNot(beNil())
+                expect(user.tokens?.refreshToken) == "hehe"
+                expect(user.tokens?.idToken) == "hehe"
+                expect(user.tokens?.userID) == "hehe"
 
-                _ = try? user.set(accessToken: "hehe again")
-                expect(newTokens).toNot(beNil())
-                newTokens = nil
+                try? user.set(refreshToken: "hehe again")
+                expect(user.tokens).toNot(beNil())
+                expect(user.tokens?.refreshToken) == "hehe again"
 
-                _ = try? user.set(refreshToken: "hehe again")
-                expect(newTokens).toNot(beNil())
-                newTokens = nil
+                try? user.set(idToken: "hehe again")
+                expect(user.tokens).toNot(beNil())
+                expect(user.tokens?.idToken) == "hehe again"
 
-                _ = try? user.set(idToken: "hehe again")
-                expect(newTokens).toNot(beNil())
-                newTokens = nil
+                try? user.set(userID: "hehe again")
+                expect(user.tokens).toNot(beNil())
+                expect(user.tokens?.userID) == "hehe again"
+            }
+        }
 
-                _ = try? user.set(userID: "hehe again")
-                expect(newTokens).toNot(beNil())
+        describe("Keychain management") {
+            it("Should not treat the user as logged in if it does not find keychain data") {
+                let user = User(state: .loggedOut)
+                expect(user.state).to(equal(UserState.loggedOut))
+            }
+
+            it("Should treat the user as logged in if it finds keychain data") {
+                Utils.createDummyKeychain()
+                let user = User(state: .loggedOut)
+                try? user.loadStoredTokens()
+                expect(user.state).to(equal(UserState.loggedIn))
+            }
+
+            it("Should update keychain when user is refreshed") {
+                self.stub(uri("/oauth/token"), try! Builders.load(file: "valid-refresh", status: 200))
+                var newTokens: TokenData?
+                do {
+                    Utils.createDummyKeychain()
+                    let user = TestingUser(state: .loggedOut)
+                    try? user.wrapped.loadStoredTokens()
+                    user.refresh { _ in }
+                    newTokens = user.wrapped.tokens
+                }
+
+                let tokens = UserTokensKeychain().data().first
+                expect(newTokens?.accessToken).to(equal(tokens?.accessToken))
+                expect(newTokens?.refreshToken).to(equal(tokens?.refreshToken))
+                expect(newTokens?.idToken).to(equal(tokens?.idToken))
```

Comment:
You can make this test more solid by getting the keychain data before the update as well and making sure they are not the equal.